core/sql/executor/ex_sort.h

```diff
@@ -356,4 +363,119 @@ class ExSortFromTopPrivateState : public ex_tcb_private_state
   ~ExSortFromTopPrivateState();       // destructor
 };
 
+class ExSortBufferPool : public NABasicObject
```

Comment:
Creating a abstract class here was considered as the first option. However, making this change would have ripple effect all over the code since sql_buffer_pool is packed and unpacked when hopping many operators. If a major change is made to sql buffer management in future, it would be nice to consider having a abstract class. For now, to keep the change concise, this approach was taken.